lib/pbench/server/indexer.py

```diff
@@ -1536,116 +1539,109 @@ def _make_source_json(self):
         We only process the top-level result.json file since it has all the
         data we need in one place.
         """
-        for dirname in self.json_dirs:
-            if dirname.startswith("sample"):
-                # sample result.json - we ignore each sample result.json since
-                # each sample is contained in the experiment level
-                # result.json.
-                continue
-            elif dirname != self.ptb.dirname:
-                # iteration result.json - there is no value in processing the
-                # result.json file at the iteration level because everything
-                # it contains is also in the experiment level's result.json.
-                continue
+        result_json = pathlib.Path(
+            self.ptb.extracted_root, self.ptb.dirname, "result.json"
+        ).resolve(strict=True)
```

Comment:
Here you don't need `.resolve(strict=True)` because you are going to `open` it below, while implies a "resolve" operation.